### PR TITLE
Remove old Room Directory

### PR DESCRIPTION
**Pull request by justjanne**
Type: Task
Fixes: https://github.com/vector-im/element-web/issues/22846

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by turt2live** [comment]
this appears to be a fairly massive change for the app, so would appreciate explicit review from the product and/or design team.

File: cypress/e2e/room-directory/room-directory.spec.ts [hunk 1]
```diff
@@ -72,32 +72,4 @@ describe("Room Directory", () => {
             expect(resp.chunk[0].room_id).to.equal(roomId);
         });
     });
-
-    it("should allow finding published rooms in directory", () => {
```
Comment (t3chguy) [inline cypress/e2e/room-directory/room-directory.spec.ts:76]: Is this case covered in the spotlight tests?

File: src/components/structures/RoomSearch.tsx [hunk 2]
```diff
@@ -44,7 +44,9 @@ export default class RoomSearch extends React.PureComponent<IProps> {
     }
 
     private openSpotlight() {
-        Modal.createDialog(SpotlightDialog, {}, "mx_SpotlightDialog_wrapper", false, true);
+        defaultDispatcher.dispatch<OpenSpotlightPayload>({
+            action: Action.OpenSpotlight,
+        });
```
Comment (t3chguy) [inline src/components/structures/RoomSearch.tsx:49]:
```suggestion
        defaultDispatcher.fire(Action.OpenSpotlight);
```

File: cypress/e2e/room-directory/room-directory.spec.ts (hunk 1 above)
Comment (justjanne) [reply cypress/e2e/room-directory/room-directory.spec.ts:76]: Finding public rooms is in spotlight's tests.

But sure, I could add a test that creates a private room, publishes it via the UI, and then tries to find it, just for completeness' sake.

**Comment by kittykat** [comment]
Are there any visual changes associated with this PR? Any risk of any remaining functionality breaking?

**Comment by turt2live** [comment]
The UI change would be the loss of the public room directory button (from the home view). It's less of a risk than what it used to be, but also still fairly significant.

**Comment by kittykat** [comment]
Changing the home screen is a more complicated issue which needs product thinking. Can that button link to this view instead of the old one?

Screenshot from 2022-10-17 22-11-06

**Comment by t3chguy** [comment]
@kittykat that is already what happens, all interactions which went to the dialog being removed are re-routed to Spotlight-in-the-explore-state

**Review by t3chguy — CHANGES_REQUESTED** [review]
Looks sane but CI is generally unhappy with it

**Review by kittykat — APPROVED** [review]
Looks fine from product pov as all links to the directory have been updated

**Pull request closed without merging**